core/linux/common.cpp

```diff
@@ -18,7 +18,7 @@
 #include <sys/param.h>
 #include <sys/mman.h>
 #include <sys/time.h>
-#if !defined(_ANDROID) && !defined(TARGET_IPHONE) && !defined(TARGET_NACL32) && !defined(TARGET_EMSCRIPTEN) && !defined(TARGET_OSX)
+#if defined(__linux__) && !defined(_ANDROID) && !defined(TARGET_IPHONE) && !defined(TARGET_NACL32) && !defined(TARGET_EMSCRIPTEN) && !defined(TARGET_OSX)
```

Comment:
Why depend on __linux__ here? Can't we reuse/introduce some other define? I generally try avoid platform-specific magic `#defines`

Reply:
Because [personality(2)](http://man7.org/linux/man-pages/man2/personality.2.html)

> personality() is Linux-specific and should not be used in programs intended to be portable.

Reply:
oh you mean like why not `HOST_OS == OS_LINUX`

well, because most of the existing code uses `OS_LINUX` for non-Linux-specific things that work on every unix